storagenode/vouchers/service.go

```diff
@@ -7,16 +7,165 @@ import (
 	"context"
 	"time"
 
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/internal/sync2"
+	"storj.io/storj/pkg/kademlia"
 	"storj.io/storj/pkg/pb"
 	"storj.io/storj/pkg/storj"
+	"storj.io/storj/pkg/transport"
+	"storj.io/storj/storagenode/trust"
+)
+
+var (
+	// VoucherError represents errors with vouchers
+	VoucherError = errs.Class("voucher")
+
+	mon = monkit.Package()
 )
 
 // DB implements storing and retrieving vouchers
 type DB interface {
 	// Put inserts or updates a voucher from a satellite
 	Put(context.Context, *pb.Voucher) error
-	// GetExpiring retrieves all vouchers that are expired or about to expire
-	GetExpiring(context.Context, time.Duration) ([]storj.NodeID, error)
 	// GetValid returns one valid voucher from the list of approved satellites
 	GetValid(context.Context, []storj.NodeID) (*pb.Voucher, error)
+	// NeedVoucher returns true if a voucher from a particular satellite is expired, about to expire, or does not exist
+	NeedVoucher(context.Context, storj.NodeID, time.Duration) (bool, error)
+}
+
+// Config defines configuration for requesting vouchers.
+type Config struct {
+	Interval         int `help:"number of days between voucher service iterations" default:"7"`
```

Comment:
if interval is 7 days then that means a new node can never be vetted faster than 7 days. Is that what we want? Seems long to me.

Reply:
can we make this be a time.Duration instead of an int plz? the default value (if 7 days) would then be "168h"

Reply:
I was torn between time.Duration for code simplicity and int for simplicity of user configuration, but changing it sounds good!